src/services/query.ts

```diff
@@ -13,6 +14,25 @@ export type LokiQuery = {
 
   datasource?: DataSourceApi;
 };
+
+/**
+ * Builds the resource query
+ * @param expr string to be interpolated and executed in the resource request
+ * @param resource
+ * @param queryParamsOverrides
+ */
+export const buildResourceQuery = (
+  expr: string,
+  resource: 'volume' | 'patterns' | 'detected_labels',
```

Comment:
We're not using this for logs? 0 context, just asking.

Reply:
Logs use `buildLokiQuery` already now - so they won't be resource calls but "usual" queries. @gtk-grafana maybe worth renaming `buildLokiQuery` to `buildDataQuery` to match the "names" we use in the datasource backend?

Reply:
Done!